### ISSUE TITLE
Richeditor - quote marks in alt tags wipes the content

### ISSUE THREAD

**Issue by rejuvenatedigital** [rainlab/translate-plugin#448]
Hi,

I've just spotted an issue where I have a repeater group that contains a richeditor field, when a user adds in an image with an alt tag that contains a quote mark for example 
> Picture of a "example"

The Froala editor then converts the quote marks to a html entity.
This then causes a problem when the reapeater group is saved in json and therefore causes the content to fail to load.

This only appears to be happening on a locale that isn't the default one.

Hope this helps.

Many thanks,
Danny

**Comment by LukeTowers** [comment]
Why would an HTML entity inside of a string cause problems in JSON? Could you post an example of your field configuration, the content you put into the Froala editor, and the generated JSON?

**Comment by rejuvenatedigital** [comment]
Hi Luke please see attached my field config, froala content and generated JSON.
[generated_json.txt](https://github.com/rainlab/translate-plugin/files/2900116/generated_json.txt)
[content_fields.txt](https://github.com/rainlab/translate-plugin/files/2900117/content_fields.txt)
[froala-markup.txt](https://github.com/rainlab/translate-plugin/files/2900118/froala-markup.txt)

**Comment by LukeTowers** [comment]
@rejuvenatedigital do it inline with the code syntax of three backticks surrounding each block of code please. I don't want to have to download and look through separate files.

**Comment by rejuvenatedigital** [comment]
@LukeTowers sorry Luke, thought it would be easier to view as files.

```
blocks:
    name: Content Blocks
    icon: icon-align-left
    fields:
        content_blocks_section:
            label: Content Block
            type: section

        content:
            label: Content
            type: richeditor

cards:
    name: Cards
    icon: icon-th-large
    fields:

        content:
            label: Content
            type: richeditor
```

```
<p>Rejuvenate Test</p>

<p><img src="/storage/app/media/1_3_320x300.jpg" style="width: 300px;" class="fr-fic fr-dib" alt="Image &quot;example&quot;"></p>

<p>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.</p>
```

```
{"name":"","slug":"bug-test","tag_line":"","description":"\r\n","hero_content":"\r\n","hero_image_alt":"","content":[{"content":"<p>Rejuvenate Test<\/p>\r\n\r\n<p><img src=\"\/storage\/app\/media\/1_3_320x300.jpg\" style=\"width: 300px;\" class=\"fr-fic fr-dib\" alt=\"Image &quot;example&quot;\"><\/p>\r\n\r\n<p>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.<\/p>\r\n","_group":"blocks"}]}
```

**Comment by LukeTowers** [comment]
@rejuvenatedigital I don't see the problem, pasting the JSON into https://jsonlint.com/ says that it's valid JSON.

**Comment by mjauvin** [comment]
@rejuvenatedigital can you create a github with your plugin code for this so we can test easily?

Also, aren't you missing the "type: repeater" in your fields definition?

**Comment by mjauvin** [comment]
I tried to reproduce and it's working fine for me.

Can we see the definition for the Model using this repeater field?

**Comment by rejuvenatedigital** [comment]
@LukeTowers I'm aware that the json is valid, the problem is when the JSON is used to generate the fields for the repeater group.

@mjauvin please find the plugin code here https://github.com/rejuvenatedigital/Property-Plugin-Example

**Comment by mjauvin** [comment]
Try renaming the "content" field in models. property/fields.yaml to
something else, I think it interferes with the field group named content
for the translatable array

--
Marc

**Comment by rejuvenatedigital** [comment]
@mjauvin just given this a try by renaming the field to **information** but the problem still occurs

**Comment by andy-laravel** [comment]
Just to clarify, this issue is still unfixed. Here's some screenshots of the issue:

Adding a new image with alt tag text with quotes:
![Screenshot 2019-03-14 at 08 16 59](https://user-images.githubusercontent.com/24649882/54341583-0fdd9600-4632-11e9-8876-6ce2e3c57d57.png)

Save entry.

Checking the DB JSON entry:
![Screenshot 2019-03-14 at 08 17 31](https://user-images.githubusercontent.com/24649882/54341604-1e2bb200-4632-11e9-87db-a554ef0d6732.png)

Reload entry in October CMS Admin:
![Screenshot 2019-03-14 at 08 17 11](https://user-images.githubusercontent.com/24649882/54341653-43b8bb80-4632-11e9-9ddb-8c47b51ba9ad.png)

**Comment by LukeTowers** [comment]
@mjauvin could you try to reproduce it?

**Comment by mjauvin** [comment]
I was able to reproduce this.

When doing the same for the DEFAULT locale, the quoted part gets removed when saving and reopening... so someone at some point added code to remove characters that were creating problems somehow... and this has not been done for the translated fields.

**Comment by mjauvin** [comment]
This is a bug in Froala's image plugin, most likely.

@daftspunk can you take a look in Froala's image plugin code for this?

**Comment by daftspunk** [comment]
This has been fixed in v3.1 since value juggling is no longer used.

- Link to the [upgrade guide](https://github.com/rainlab/translate-plugin/blob/master/UPGRADE.md)
- Link to [multisite documentation](https://docs.octobercms.com/3.x/cms/resources/multisite.html)

If you still need help, feel free to get in touch with the support desk for assistance: https://octobercms.com/contact
